src/perform_step/rkc_perform_step.jl

```diff
@@ -768,7 +773,7 @@ function perform_step!(integrator, cache::IRKCCache, repeat_step=false)
   if isnewton(nlsolver) && integrator.opts.adaptive
     update_W!(integrator, cache, dt, false)
     @.. gprev = dt*0.5*(du₂ - f2ⱼ₋₁) + dt*(0.5 - μs₁)*(du₁ - f1ⱼ₋₁)
-    nlsolver.linsolve(vec(tmp),get_W(nlsolver),vec(gprev),false)
+    get_linsolve(nlsolver)(vec(tmp),get_W(nlsolver),vec(gprev),false)
```

Comment:
This feels pretty bad to me (appears in multiple other places as well). I think we should rather define a generic method `linsolve(out, nlsolver, in, flag; kwargs...)` (or maybe some other less generic name) (in DiffEqBase?) that's defined as `f_linsolve = get_linsolve(nlsolver); W = get_W(nlsolver); f_linsolve(vec(out), W, vec(in), flag; kwargs...)`.

Reply:
yeah that makes sense.